lib/modules/dosomething/dosomething_signup/dosomething_signup.module

```diff
@@ -206,8 +206,8 @@ function dosomething_signup_get_query_source() {
  *   The sid of the newly inserted signup, or FALSE if error.
  */
 function dosomething_signup_create($nid, $uid = NULL, $source = NULL, $timestamp = NULL) {
+  global $user;
   if ($uid == NULL) {
-    global $user;
```

Comment:
oh yeah andrea's right here, this is conflicting with the param usage. ill fix fix